site/src/job_queue/mod.rs

```diff
@@ -278,28 +291,89 @@ async fn process_benchmark_requests(
             }
         }
     }
-    Ok(())
+    Ok(completed)
 }
 
 /// For queueing jobs, add the jobs you want to queue to this function
-async fn cron_enqueue_jobs(site_ctxt: &SiteCtxt) -> anyhow::Result<()> {
-    let mut conn = site_ctxt.conn().await;
+async fn cron_enqueue_jobs(ctxt: &SiteCtxt) -> anyhow::Result<()> {
+    let mut conn = ctxt.conn().await;
 
-    let index = conn.load_benchmark_request_index().await?;
+    let index = ctxt.known_benchmark_requests.load();
 
+    let mut requests_inserted = false;
     // Put the master commits into the `benchmark_requests` queue
-    create_benchmark_request_master_commits(site_ctxt, &*conn, &index).await?;
+    requests_inserted |= create_benchmark_request_master_commits(ctxt, &*conn, &index).await?;
     // Put the releases into the `benchmark_requests` queue
-    create_benchmark_request_releases(&*conn, &index).await?;
+    requests_inserted |= create_benchmark_request_releases(&*conn, &index).await?;
```

Comment:
If we have created `master` commits using `|=` will mean we will not create release commits meaning we potentially miss adding elements to the queue. If I'm reading it correctly;
```rs
requests_inserted |= create_benchmark_request_master_commits(ctxt, &*conn, &index).await?;
```
Evaluates to `true`. Then as `requests_inserted` is now true it will short circuit and not call `create_benchmark_request_releases`?

Though we are not wanting to insert release commits if we insert master this comment can be ignored.

Reply:
`|` (bit or) is not short-circuiting in Rust, same as in C (https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=25f0365af521a2da542fe47bb1e07be1) :)